src/xref2/tools.ml

```diff
@@ -585,9 +613,10 @@ and lookup_and_resolve_module_from_path :
     | `Resolved (`Identifier i as resolved_path) ->
         let m = Env.lookup_module i env in
         return (process_module_path env add_canonical m resolved_path, m)
-    | `Resolved r ->
-        let m = lookup_module env r in
-        return (process_module_path env add_canonical m r, m)
+    | `Resolved r -> (
+        match lookup_module env r with
+        | Ok m -> return (process_module_path env add_canonical m r, m)
+        | Error _ -> Unresolved (`Resolved r) )
```

Comment:
!

Reply:
`` `Resolved r `` is basically `p`. Is this wrong ? I removed the weird wording

Reply:
Just the idea of returning `Unresolved (Resolved r)` stands out. It's fine though, I don't see there's any other sensible value to return.